Code/Tools/FBuild/FBuildCore/BFF/Functions/FunctionPrint.cpp

```diff
@@ -79,6 +79,60 @@ FunctionPrint::FunctionPrint()
 
             FLOG_BUILD_DIRECT( tmp.Get() );
         }
+        else if ( c == '/' )
+        {
+            // find end of pattern
+            BFFIterator stop( start );
+            do
+            {
+              stop++;
+            } while ( stop.GetCurrent() < functionHeaderStopToken->GetCurrent() && *stop != c );
+            if ( stop.GetCurrent() == functionHeaderStopToken->GetCurrent() )
+            {
+              Error::Error_1002_MatchingClosingTokenNotFound( start, nullptr, c );
+              return false;
+            }
+
+            // perform variable substitutions
+            AStackString< 1024 > pat;
+        
+            start++; // skip past opening quote
+            if ( BFFParser::PerformVariableSubstitutions( start, stop, pat ) == false )
+            {
+                return false; // substitution will have emitted an error
+            }
+
+            stop++;
+            stop.SkipWhiteSpaceAndComments();
+            
+            // Look at the options.
+            bool iterateParents = false;
+            bool shouldRecurse = false;
+        
+            for ( BFFIterator iterOpt = stop; stop.GetCurrent() < functionHeaderStopToken->GetCurrent(); )
+            {
+                switch ( *iterOpt )
+                {
+                    case 'P':
+                        iterateParents = true;
+                        break;
+                    case '+':
+                        shouldRecurse = true;
+                        break;
+                    default:
+                    {
+                        AStackString<32> err;
+                        err.Format("Print pattern option %c", *iterOpt );
+                        Error::Error_1030_UnknownDirective(iterOpt, err);
+                        return false;
+                    }
+                }
+                stop++;
+                stop.SkipWhiteSpaceAndComments();
```

Comment:
I put this in to consume whitespace between the options and the close parens.  I'm ok with it allowing whitespace between the options, even though that's not "standard"